quattor/types/metadata.pan

```diff
@@ -14,7 +14,35 @@ type structure_template = {
     "branch" : structure_branch
 };
 
+type structure_os_version = {
+    @{Descriptive form of version, e.g. 7x}
+    "name" : string
+    @{Major version number component, e.g. 7}
+    "major" ? long(0..)
+    @{Minor version number component}
+    "minor" ? long(0..)
+};
+
+type structure_os_distribution = {
+    @{Distribution name, e.g. sl}
+    "name" : string with match(SELF, '^((rh)?el|centos|sl)$')
+    @{Full descriptive name, e.g. Scientific Linux}
+    "description" ? string
+    @{Family that this distribution belongs to e.g. The family for "sl" would be "el"}
+    "family" ? string with match(SELF, '^el$')
+};
+
+type structure_os = {
+    @{OS distribution details}
+    "distribution" : structure_os_distribution
+    @{OS version details}
+    "version" : structure_os_version
+    @{Architecture of OS (may be different to system architecture)}
+    "architecture" : string
+};
+
 type structure_metadata = {
     "template" : structure_template
     "features" ? string[]
+    "os" ? structure_os
```

Comment:
I'm quite happy to move this under `/system/os`, I don't believe it belongs under `archetype` as it is not archetype specific, I am also concious that you have expressed concern in the past about using `/system` as a dumping ground for information.

I've never been clear what the rules are around `/metadata` in my mind that is what this is (since it is informational), so we definitely should document the philosophy of the schema sections.

`/system/archetype/os` has never contained anything for us, are you sure the broker uses it?